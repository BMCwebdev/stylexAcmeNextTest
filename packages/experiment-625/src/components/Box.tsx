import { ReactNode } from 'react';
import { colors } from '@bmcwebdev/stitch-tokens/colors.stylex';
import { sizes } from '@bmcwebdev/stitch-tokens/sizes.stylex';

import * as stylex from '@stylexjs/stylex';

const styles = stylex.create({
  box: {
    background: colors.neutral_0,
  },
  border: {
    border: `${sizes.border_md} solid ${colors.neutral_500}`,
  },
  variantRed: {
    borderColor: colors.red_500,
    background: colors.red_50,
    color: colors.red_900,
  },
  variantGreen: {
    borderColor: colors.green_500,
    background: colors.green_50,
    color: colors.green_900,
  },
  variantBlue: {
    borderColor: colors.blue_500,
    background: colors.blue_50,
    color: colors.blue_900,
  },
  variantYellow: {
    borderColor: colors.yellow_500,
    background: colors.yellow_50,
    color: colors.yellow_900,
  },
  paddingSmall: {
    padding: sizes.spacing_100,
  },
  paddingMedium: {
    padding: sizes.spacing_400,
  },
  paddingLarge: {
    padding: sizes.spacing_700,
  },
  radiusSmall: {
    borderRadius: sizes.radius_sm,
  },
  radiusMedium: {
    borderRadius: sizes.radius_md,
  },
  radiusLarge: {
    borderRadius: sizes.radius_lg,
  },
});

export interface BoxProps {
  border?: boolean;
  children: ReactNode;
  padding?: 'none' | 'small' | 'medium' | 'large';
  radius?: 'none' | 'small' | 'medium' | 'large';
  variant?: 'blue' | 'default' | 'green' | 'red' | 'yellow';
}

export function Box({
  border,
  children,
  padding = 'medium',
  radius = 'none',
  variant = 'default',
}: BoxProps) {
  return (
    <div
      {...stylex.props(
        styles.box,
        border && styles.border,
        variant === 'red' && styles.variantRed,
        variant === 'green' && styles.variantGreen,
        variant === 'blue' && styles.variantBlue,
        variant === 'yellow' && styles.variantYellow,
        radius === 'small' && styles.radiusSmall,
        radius === 'medium' && styles.radiusMedium,
        radius === 'large' && styles.radiusLarge,
        padding === 'small' && styles.paddingSmall,
        padding === 'medium' && styles.paddingMedium,
        padding === 'large' && styles.paddingLarge,
      )}
    >
      {children}
    </div>
  );
}
