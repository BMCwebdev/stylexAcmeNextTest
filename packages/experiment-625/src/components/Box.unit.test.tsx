import { describe, expect, test } from 'vitest';
import { render, screen } from '@testing-library/react';

import { Box } from './Box';

describe('Box', async () => {
  test('renders a basic box', async () => {
    const { container } = render(<Box>Box content</Box>);
    expect(screen.getByText('Box content')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
  test('renders a box with a border', async () => {
    const { container } = render(<Box border>Box content</Box>);
    expect(screen.getByText('Box content')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
  test('renders a box with a border, radius, and large padding', async () => {
    const { container } = render(
      <Box border radius="medium" padding="large">
        Box content
      </Box>,
    );
    expect(container).toMatchSnapshot();
  });
  test('renders a box with a colored variant', async () => {
    const { container } = render(
      <Box border variant="blue">
        Box content
      </Box>,
    );
    expect(container).toMatchSnapshot();
  });
});
