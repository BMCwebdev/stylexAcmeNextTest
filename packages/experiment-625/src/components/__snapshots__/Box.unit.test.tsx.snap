// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Box > renders a basic box 1`] = `
<div>
  <div
    class="Box__styles.box xk3s6z2 Box__styles.paddingMedium x3pnyd4"
  >
    Box content
  </div>
</div>
`;

exports[`Box > renders a box with a border 1`] = `
<div>
  <div
    class="Box__styles.box xk3s6z2 Box__styles.border x10n4xv1 Box__styles.paddingMedium x3pnyd4"
  >
    Box content
  </div>
</div>
`;

exports[`Box > renders a box with a border, radius, and large padding 1`] = `
<div>
  <div
    class="Box__styles.box xk3s6z2 Box__styles.border x10n4xv1 Box__styles.radiusMedium xyjsr0 Box__styles.paddingLarge x1fzn9rj"
  >
    Box content
  </div>
</div>
`;

exports[`Box > renders a box with a colored variant 1`] = `
<div>
  <div
    class="Box__styles.box Box__styles.border x10n4xv1 Box__styles.variantBlue x1fcakp xbsrw2h xc9984p Box__styles.paddingMedium x3pnyd4"
  >
    Box content
  </div>
</div>
`;
